determined-ai/determined
# fix: do not publish helm chart for *-rc releases [DPS-260]

## Description

fix: do not publish helm chart for *-rc releases [DPS-260]

Release runs in CI publish the helm chart in a new GitHub release for *-rc builds. We do not want these. This PR extracts the helm chart GH release magic into a new CI job with a non-rc release filter.

## Test Plan

1. Somehow trigger the `release` workflow in CircleCI on a `x.y.z` tag in a separate repository.
2. Verify helm chart is in newly published release artifacts.
3. trigger `release` workflow again on a `x.y.z-rcN` tag in a separate repository.
4. verify no new release is created for this tag.

## Commentary

Some commits are NOT intended to be merged. These adjust various CI configs to facilitate testing and must be reverted before merging this PR.

UPDATE: the commits are reverted in 91994d9882d7550df21862a0e952e27dd2f46617

## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.


[DPS-260]: https://determinedai.atlassian.net/browse/DPS-260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
